gcovr/gcov.py

```diff
@@ -184,6 +185,7 @@ def guess_source_file_name(
         fname = guess_source_file_name_heuristics(
             gcovname, data_fname, currdir, root_dir, starting_dir, obj_dir, gcda_fname
         )
+    fname = str(pathlib.Path(fname).resolve())  # resolves case insensitively of windows paths
```

Comment:
This also resolves symlinks which is a big change here.

Reply:
If `os.path.realpath` doesn't fix the case we should first replace every usage of `os.path` with `pathlib.Path`.

Reply:
I tried to replace `os.path` with `pathlib.Path` but stopped this because there are differences under the hood on Windows which results in Exceptions if used in SCM Clearcase.
Win errors allowed in os.path https://github.com/python/cpython/blob/a87c46eab3c306b1c5b8a072b7b30ac2c50651c0/Lib/ntpath.py#L615-L629 and here the errors allowed in pathlib https://github.com/python/cpython/blob/4710a7589aee48bec9099af308bf5938285ba336/Lib/pathlib.py#L33-L36.

Reply:
In that case simply using ``os.path.realpath`` instead of ``pathlib.Path().resolve()`` seam to be the better solution.

Reply:
But this also resolves the symlinks. I think it would be better fix the casing on our own.

Reply:
If we want to fix the casing by our self, I think to only reasonable soltuion would be to recursivlely resolve the path and check if the case of the child fits the resolved path.
An alternative easy soltuion would be to convert all paths to lower case if the OS is case insensitve.

Reply:
I prefer the first solution by splitting the path and fix the case for each part.